supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/actors/GcpBatchBackendSingletonActor.scala

```diff
@@ -1,126 +1,66 @@
-      }.onComplete {
-        case Failure(exception) =>
-          log.error(exception, s"Failed to submit job (${request.jobName}) to GCP, workflowId = ${request.workflowId}")
-          replyTo ! Event.ActionFailed(request.jobName, exception)
-
-        case Success(job) =>
-          log.info(s"Job (${request.jobName}) submitted to GCP, workflowId = ${request.workflowId}, id = ${job.getUid}")
-          replyTo ! Event.JobSubmitted(job)
-      }
-
-    case Action.QueryJob(jobName) =>
-      val replyTo = sender()
-
-      Future {
-        requestHandler.query(requestFactory.queryRequest(jobName))
-      }.onComplete {
-        case Success(job) =>
-          log.info(s"Job ($jobName) retrieved from GCP, state = ${job.getStatus.getState}")
-          replyTo ! Event.JobStatusRetrieved(job)
-
-        case Failure(exception) =>
-          log.error(exception, s"Failed to query job status ($jobName) from GCP")
-          replyTo ! Event.ActionFailed(jobName.toString, exception)
-      }
-
-    case Action.AbortJob(jobName) =>
-      val replyTo = sender()
-
-      Future {
-        requestHandler.abort(requestFactory.abortRequest(jobName))
-      }.onComplete {
-        case Success(operation) =>
-          log.info(s"Job ($jobName) aborted from GCP")
-          replyTo ! Event.JobAbortRequestSent(operation)
-
-        case Failure(exception) =>
-          log.error(exception, s"Failed to abort job ($jobName) from GCP")
-          replyTo ! Event.ActionFailed(jobName.toString, exception)
-      }
-
-    // Cromwell sends this message
-    case BackendSingletonActorAbortWorkflow(workflowId) =>
-      // It seems that AbortJob(jobName) is processed before this message, hence, we don't need to do anything else.
-      // If it ever becomes necessary, we'll need to create link submitted jobs to its workflow id, which require
+    with ActorLogging {
+
+  private val jesApiQueryManager = context.actorOf(
+    BatchApiRequestManager
+      .props(qps = qps, requestWorkers = requestWorkers, serviceRegistryActor, batchRequestExecutor)
+      .withMailbox(Mailbox.PriorityMailbox),
+    "BatchApiRequestManager"
+  )
+
+  override def receive = {
+    case _: BackendSingletonActorAbortWorkflow =>
+      // Cromwell sends this message to abort a job, but "GcpBatchAsyncBackendJobExecutionActor" implements
+      // the "tryAbort" method which makes this message irrelevant.
+      //
+      // If it ever becomes necessary, we'll need to link submitted jobs to its workflow id, which require
       // us to be cautious because batch deletes jobs instead of canceling them, hence, we should not delete jobs
       // that are on a final state.
-      log.info(s"Cromwell requested to abort workflow $workflowId")
+      //
+      // jesApiQueryManager.forward(abort)
+      ()
+
+    case apiQuery: BatchApiRequest =>
+      log.debug("Forwarding API query to Batch request manager actor")
+      jesApiQueryManager.forward(apiQuery)
```

Comment:
Legacy naming: not a super high priority, but "JES" stands for "Job Execution Service" and is an ancestor of Batch.

JES -> Pipelines API v1 -> Pipeines API v2alpha -> Pipeines API v2beta -> Genomics -> Life Sciences -> Batch

Reply:
I have resolved this.